mesonbuild/linkers/linkers.py

```diff
@@ -1317,6 +1317,9 @@ def get_always_args(self) -> T.List[str]:
     def get_win_subsystem_args(self, value: str) -> T.List[str]:
         return self._apply_prefix([f'/SUBSYSTEM:{value.upper()}'])
 
+    def fatal_warnings(self) -> T.List[str]:
+        return ['-WX']
+
```

Comment:
Seems like these are unrelated changes? Might make sense to split them out into their own commit

Reply:
No, they're part of the same. See this project I'm testing it with: https://github.com/dragonCodecs/libusb/blob/blackmagic/meson/meson.build#L279

With the `-Wl` conversion, the `-WX` flag can make it past clang into the LINK.exe line. And `-WX` turns the warning

> unrecognized option '/-add-stdcall-alias'; ignored 

into a hard failure, which is what Meson uses for detecting unsupported flags.

For the specific error, see https://learn.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4044?view=msvc-170

Reply:
No? It's necessary afaict: https://learn.microsoft.com/en-us/cpp/build/reference/wx-treat-linker-warnings-as-errors?view=msvc-170

Reply:
Sorry, my browser didn't load amy's comment.